frontend/lib/src/components/shared/Modal/Modal.tsx

```diff
@@ -135,8 +136,41 @@ const ModalButton: FunctionComponent<
   </StyledModalButton>
 )
 
-function Modal(props: ModalProps): ReactElement {
-  const { spacing, radii, colors }: EmotionTheme = useTheme()
+export type StreamlitModalProps = Omit<ModalProps, "size"> & {
```

Comment:
note: in context of refactoring the width calculation so that it can be shared between the DeployDialog and the Dialog.tsx, I have introduced this type to abstract the Baseui size away. This way, our components only deal with the Streamlit-defined size which would make it easier to swap out Baseweb in the future